tests/attention/test_cute_dsl_decode.py

```
@@ -123,6 +129,368 @@ def _decode_reference_contiguous(
+
+    wrapper.plan("indptr", "indices", "last", 32, 4, 128, 16, window_right=4)
+    assert captured["kwargs"]["window_right"] == 4
+
+    with pytest.raises(TypeError, match="at most 16"):
+        wrapper.plan("indptr", "indices", "last", 32, 4, 128, 16, *(None,) * 17)
+
+
+def test_batch_decode_plan_duplicate_deprecated_positional_arg_rejected(monkeypatch):
+    wrapper, _ = _capture_decode_plan_impl(monkeypatch)
+
+    with pytest.raises(TypeError, match="multiple values.*pos_encoding_mode"):
```

Ruff (0.15.20)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)